hcl/writer.go

```diff
@@ -203,3 +212,25 @@ func walk(dest, src reflect.Value, interpolate map[string]string, name string, r
 		dest.Set(src)
 	}
 }
+
+// isMutualInterpolation will simply go through the list of relations to find out
+// if a relation is already present between the two resources
+func isMutualInterpolation(target, source string, relations *[][]string) bool {
+	for _, r := range *relations {
+		if (r[0] == target &&
+			r[1] == source) ||
+			(r[1] == target &&
+				r[0] == source) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I think it would be better to have a `map[string]struct{}` where the key is `source+target`, so we can easily check without iterating over all the array.

Also document what is what.

Reply:
it's documented :thinking: 
```go
// isMutualInterpolation will simply go through the list of relations to find out
// if a relation is already present between the two resources
```